src/telemetry/logs.ts

```diff
@@ -5,7 +5,22 @@ import { Logger as OTelLogger, LogAttributes, SeverityNumber } from '@openteleme
 import { LogRecord, LoggerProvider } from '@opentelemetry/sdk-logs';
 import { Span } from '@opentelemetry/sdk-trace-base';
 import { TelemetryCollector } from './collector';
-import { DBOSJSON, globalAppVersion } from '../utils';
+import { DBOSJSON, globalAppVersion, CircularBuffer, interceptStreams } from '../utils';
+
+// Capture the native stdout and stderr write functions
+// We will override them twice: once for buffering logs and once for forwarding logs to the DBOS global logger
+const logBuffer = new CircularBuffer(100);
```

Comment:
This always runs, even if OTLP logging isn't configured. I'm very nervous about this code (and the code in `utils.ts`, especially in unusual environments like Next.js. I think a much safer way to do this is to totally eliminate the buffer part and configure interception only if OTLP is setup (and so only after `DBOS.launch()`. That means OTLP doesn't get logs from before `launch`, but that's fine for a first pass.